src/main.c

```diff
@@ -3293,6 +3376,13 @@ H2O_NORETURN static void *run_loop(void *_thread_index)
             conf.listeners[i]->quic.thread_fds[thread_index] = fds[1];
         }
     }
+
+    /* setup the h2olog listener */
+    if (conf.h2olog.listen_fd != -1) {
```

Comment:
```suggestion
    if (thread_index == 0 && conf.h2olog.listen_fd != -1) {
```
We only need to have one thread listening to h2olog (and we use thread zero for such purposes, see below).